### PR TITLE
Command argument types for `drep` commands

### PR DESCRIPTION
**Pull request by newhoggy**
# Changelog

```yaml
- description: |
    Command argument types for `drep` commands
# uncomment types applicable to the change:
  type:
  # - feature        # introduces a new feature
  # - breaking       # the API has changed in a breaking way
  # - compatible     # the API has changed but is non-breaking
  # - optimisation   # measurable performance improvements
  - improvement    # QoL changes e.g. refactoring
  # - bugfix         # fixes a defect
  # - test           # fixes/modifies tests
  # - maintenance    # not directly related to the code
  # - release        # related to a new release preparation
  # - documentation  # change in code docs, haddocks...
```

# Context

# How to trust this PR

Highlight important bits of the PR that will make the review faster. If there are commands the reviewer can run to observe the new behavior, describe them.

# Checklist

- [ ] Commit sequence broadly makes sense and commits have useful messages
- [ ] New tests are added if needed and existing tests are updated. See [Running tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- [ ] Self-reviewed the diff

### CODE REVIEW

File: cardano-cli/src/Cardano/CLI/EraBased/Run/Governance/DRep.hs [hunk 1]
```diff
@@ -16,54 +18,74 @@ import           Cardano.Api.Ledger (Credential (KeyHashObj))
...
-  => ConwayEraOnwards era
-  -> VerificationKeyOrFile DRepKey
-  -> IdOutputFormat
-  -> Maybe (File () Out)
+  => Cmd.GovernanceDRepIdCmdArgs era
   -> ExceptT GovernanceCmdError IO ()
-runGovernanceDRepIdCmd _ vkOrFp idOutputFormat mOutFile = do
+runGovernanceDRepIdCmd
+    Cmd.GovernanceDRepIdCmdArgs
+      { eon = _
```
Comment (carbolymer) [inline cardano-cli/src/Cardano/CLI/EraBased/Run/Governance/DRep.hs]:
```suggestion
      {
```
I think it's not needed here.

File: cardano-cli/src/Cardano/CLI/EraBased/Run/Governance/DRep.hs [hunk 2]
```diff
@@ -78,54 +100,64 @@ runGovernanceDRepIdCmd _ vkOrFp idOutputFormat mOutFile = do
...
-  :: DRepMetadataFile In
-  -> Maybe (File () Out)
+runGovernanceDRepMetadataHashCmd :: ()
+  => Cmd.GovernanceDRepMetadataHashCmdArgs era
   -> ExceptT GovernanceCmdError IO ()
-runGovernanceDRepMetadataHashCmd drepMDPath mOutFile = do
-  metadataBytes <- firstExceptT ReadFileError $ newExceptT (readByteStringFile drepMDPath)
+runGovernanceDRepMetadataHashCmd
+    Cmd.GovernanceDRepMetadataHashCmdArgs
+      { eon = _
```
Comment (carbolymer) [inline cardano-cli/src/Cardano/CLI/EraBased/Run/Governance/DRep.hs]:
```suggestion
      {
```
I think it's not needed here

**Review by carbolymer — APPROVED** [review]

File: cardano-cli/src/Cardano/CLI/EraBased/Run/Governance/DRep.hs (hunk 1 above)
Comment (newhoggy) [reply cardano-cli/src/Cardano/CLI/EraBased/Run/Governance/DRep.hs]: Remove

File: cardano-cli/src/Cardano/CLI/EraBased/Run/Governance/DRep.hs (hunk 2 above)
Comment (newhoggy) [reply cardano-cli/src/Cardano/CLI/EraBased/Run/Governance/DRep.hs]: Remove

**Pull request merged**